### PR TITLE
*: unify the usage of independent service

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: ref #8477.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -2439,9 +2439,25 @@ func IsClientURL(addr string, etcdClient *clientv3.Client) bool {
 
 // IsServiceIndependent returns whether the service is independent.
 func (c *RaftCluster) IsServiceIndependent(name string) bool {
-	independent, exist := c.independentServices.Load(name)
-	if !exist {
+	if c == nil {
 		return false
 	}
-	return independent.(bool)
+	_, exist := c.independentServices.Load(name)
```
Comment (lhy1024) [inline server/cluster/cluster.go:2445]: Do we need not to check the result?

Comment (rleungx) [reply server/cluster/cluster.go:2445]: no

**Review by lhy1024 — APPROVED** [review]

File: pkg/utils/apiutil/serverapi/middleware.go [hunk 2]
```diff
@@ -129,7 +129,7 @@ func (h *redirector) matchMicroServiceRedirectRules(r *http.Request) (bool, stri
 	for _, rule := range h.microserviceRedirectRules {
 		// Now we only support checking the scheduling service whether it is independent
 		if rule.targetServiceName == mcsutils.SchedulingServiceName {
-			if !h.s.IsServiceIndependent(mcsutils.SchedulingServiceName) {
+			if !h.s.GetRaftCluster().IsServiceIndependent(mcsutils.SchedulingServiceName) {
```
Comment (HuSharp) [inline pkg/utils/apiutil/serverapi/middleware.go:132]: Do we need to unify `mcsutils.SchedulingServiceName` and `utils.SchedulingServiceName` as well?

Comment (rleungx) [reply pkg/utils/apiutil/serverapi/middleware.go:132]: https://github.com/tikv/pd/pull/8476 will change it to constant.xxx

**Review by HuSharp — APPROVED** [review]

**Pull request merged**